### ISSUE TITLE
GTMetrix is no longer usable without account

### ISSUE THREAD

**Issue by garritfra** [kevquirk/512kb.club#1366]
Just stumbled across this by reviewing a site:

<img width="824" alt="grafik" src="https://github.com/kevquirk/512kb.club/assets/32395585/a571abfa-79e9-4394-acce-9f0e04272b7c">

Looks like we have to switch. [Pingdom](https://tools.pingdom.com/) maybe?

**Comment by garritfra** [comment]
Pingdom unfortunately has different results for page sizes. Examples:

<details>
<summary>fanrongbin.com</summary>

GTMetrix: 184 kB

<img width="263" alt="Bildschirmfoto 2023-12-14 um 13 13 46" src="https://github.com/kevquirk/512kb.club/assets/32395585/a293ea1c-1281-4bb3-9ff7-f62439ab2e66">

Pingdom: 97.5 kB

<img width="1028" alt="Bildschirmfoto 2023-12-14 um 13 13 25" src="https://github.com/kevquirk/512kb.club/assets/32395585/2714f697-5cd5-4a70-9b15-e9f7bf447a43">

</details>


<details>
<summary>www.blitzw.in</summary>

GTMetrix: 21 kB

<img width="343" alt="Bildschirmfoto 2023-12-14 um 13 13 36" src="https://github.com/kevquirk/512kb.club/assets/32395585/6d2129c8-0cdf-4f23-b6b3-68dcb6aa1ae8">


Pingdom: 23.5 kB

<img width="1030" alt="Bildschirmfoto 2023-12-14 um 13 12 36" src="https://github.com/kevquirk/512kb.club/assets/32395585/c022d3bb-5c8f-45f6-a5aa-a2c2ab46e3c9">

</details>

**Comment by kevquirk** [comment]
Oh damn. Might need to do some research as to what we can use.

**Comment by garritfra** [comment]
FWIW, I checked if https://pagespeed.web.dev/ might be an alternative, but they don't seem to expose page weight at all. 🤔

**Comment by kevquirk** [comment]
I don't think that will work as it doesn't mention page size.

1mb.club is using [DebugBear](https://www.debugbear.com/test/website-speed) doing a test and comparing it to GT Metrix, it comes our with different scores though. 🤷‍♂️

**Comment by garritfra** [comment]
DebugBear seems to be quite close to the compressed size reported by GTMetrix. If all else fails, that might be our only option. 🤷‍♂️

**Comment by kevquirk** [comment]
Yeah, but it means we will need to re-do over 1000 sites to make sure it's accurate. 😳

**Comment by kevquirk** [comment]
I'm wondering if we take this as serendipity and close the project down? I don't have the time to go through all these sites and re-test them.

What do you think, @garritfra ?

**Comment by garritfra** [comment]
I wouldn't want to close the project down per se. I'll contact you to discuss some details.

**Comment by meduzen** [comment]
> I'm wondering if we take this as serendipity and close the project down?

I was coming to update a size and stumbled upon the same issue. Closing the project could be a nice testimony from the (recent) past, but at the same time it’s also kinda useful to discover new things.

Without automation, maintenance efficiency is kinda limited.

**Comment by michaelnordmeyer** [comment]
Some services include the reply headers in the reported weight, some don't. You can use `curl` to find out, but you only get the data for the HTML page.

With compression:

```sh
curl --compressed -so /dev/null -w "Header: %{size_header} bytes, Download: %{size_download} bytes\n" https://kevquirk.com/
```

```text
Header: 259 bytes, Download: 3814 bytes
```

Without compression:

```sh
curl -so /dev/null -w "Header: %{size_header} bytes, Download: %{size_download} bytes\n" https://kevquirk.com/ 
```

```text
Header: 236 bytes, Download: 31883 bytes
```

`wget`, as someone mentioned in https://github.com/kevquirk/512kb.club/pull/749, would be an easy solution, but it doesn't work for CSS files importing other files.

**Comment by derspyy** [comment]
wow this whole stuff sucks : /
i wholeheartedly think the project should continue at least with compressed sizes.
it was always awesome to see what could be done!

**Comment by JLO64** [comment]
I know this isn't productive towards the current conversation, but it gets worse with GTmetrix. Even if you sign up for a free account there is a limit for how many reports you can generate. After you hit that limit (I'm not sure what it is) you are forced to pay for a plan in order to generate more.

![image](https://github.com/kevquirk/512kb.club/assets/50301800/3ff522eb-f5b3-47e1-a8f2-9e1918633069)

Regarding shutting down the project, I'd argue against it. It's genuinely nice having a list of websites like these in a format like this. (I'm totally not in it for the ) Maybe it can be restarted instead with a new list starting from scratch based on whatever new metric is agreed upon? As nice as debugbear is I think it would be wise to avoid third party services to avoid another situation like what we're going through now.

**Comment by garritfra** [comment]
I think there's no way around a third party service like debug bear to get consistent results.

We won't shut down the project. Once we find a solution we're happy with, updating all sites shouldn't be an issue using scripts. Ideally we should try to collaborate with the third party service owner (whoever that might be) to lift any possible scan restrictions or speed up the process.

**Comment by bradleytaunt** [comment]
Thought I would throw my two cents in:

For [1mb.club](https://1mb.club) I have been toying with the idea of using a custom script that will: 

1. Download all the files of a given website (CSS, images, JS, main HTML page)
2. Place all these files into a temporary folder
3. Get the size of all those files together as the total

I finally pushed out what I working on way back and called it "sizegrab": https://git.sr.ht/~bt/sizegrab (it's written in ruby)

It's far from perfect and most likely doesn't cover every use case. But the idea is to try and avoid depending on third party companies / services. If anyone wants to help improve that *ugly* script of mine, please do so! (or even re-write everything in a language others prefer!)

/ end rant

**Comment by JLO64** [comment]
> It's far from perfect and most likely doesn't cover every use case. But the idea is to try and avoid depending on third party companies / services. If anyone wants to help improve that _ugly_ script of mine, please do so! (or even re-write everything in a language others prefer!)

I am probably the least qualified person in this thread to be working on something like this, but I have a bunch of free time on my hands I'm willing to put towards this. For right now, I'm thinking of rewriting this as a Python script (I'm sorry I'm not more comfortable with Ruby!) and deploying it as an AWS Lambda function. After I get that set up, I can make a widget and/or website to query and display results. Since it'll be a Lambda function, maybe it can be used to better automate the project in the future?

**Comment by evtn** [comment]
> FWIW, I checked if https://pagespeed.web.dev/ might be an alternative, but they don't seem to expose page weight at all. 🤔

Actually they do, under `Performance -> Passed Audits -> Avoids enormous network payloads`

Although I'm not sure if it's a size after compression

**Comment by michaelnordmeyer** [comment]
> Although I'm not sure if it's a size after compression

It's the size after compression. You can test with your browser's developer tools on the network tab. They display both sizes.

**Comment by JLO64** [comment]
Lol. While working on [a solution of my own](https://whimsical-nasturtium-f1effa.netlify.app/) I stumbled across [Cloudflare's URL Scanner](https://radar.cloudflare.com/scan/56af38c6-6cfd-494a-ab8a-20cbffddfe5f/summary). It's pretty robust, free to use with no restrictions, and has an API (you have to sign up and generate a token to use the API tho)
![image](https://github.com/kevquirk/512kb.club/assets/50301800/daadb443-5ed9-49ca-804a-763042618000)

They even show both compressed and uncompressed network transfers!
![image](https://github.com/kevquirk/512kb.club/assets/50301800/0bc6b954-02a1-4cc5-bb8a-18075874b752)

**Comment by garritfra** [comment]
@JLO64 huh, nice! I'm not sure if our audience will be kosher using cloudflare, but I'd personally be fine with this. @kevquirk what do you think?

**Comment by kevquirk** [comment]
I just checked on my site and the results are very close. GTMetrix is reporting 89.4kb and Cloudflare is reporting 91.5kb.

While 2kb is quite a lot when it comes to the smaller sites, it's about the closest we've come out of everything we've tried. Being Cloudflare, it's unlikely to go paid anytime soon too. Also, having the API, we could potentially go back and re-test all sites with CF (although I'd have no idea how to do that haha).

TL;DR I think we have a winner. If you're happy @garritfra we can update the instructions on the site.

**Comment by garritfra** [comment]
@kevquirk I'll take care of it.

Thanks @JLO64 for the suggestion!

**Comment by garritfra** [comment]
#1383 is merged, so I'll close this issue. Thanks for all the input!

Any contributions regarding the [automatic size checker](https://github.com/kevquirk/512kb.club/blob/main/scripts/site_size_rechecker.py) or other topics discussed here are welcome. I don't think I'll have the time to rewrite that script any time soon.
